amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -357,5 +359,12 @@ public ActionForward viewDifferences(ActionMapping mapping, ActionForm form, Htt
 
         return mapping.findForward("forward");
     }
-
+    
+    public ActionForward viewAllDifferences(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        Map<Long, Map<String, List<CompareOutput>>> listoutputCollectionGrouped = new HashMap<Long, Map<String, List<CompareOutput>>>();
+         listoutputCollectionGrouped = ActivityVersionUtil.compareActivities(AuditLoggerUtil.getListOfActivitiesFromAuditLogger());        
+         vForm.setListoutputCollectionGrouped(listoutputCollectionGrouped);
```

Comment:
vForm.setListoutputCollectionGrouped is populating the internal variable listoutputCollectionGrouped and then the jsp file is using   return this.outputCollectionGrouped.entrySet(); which is null. also the return type of the entryset is not consistent. I think you have might have code missing to commit since it always returns nullpointer and Im sure for you worked